src/app/todo-list/todo-list.reducer.ts

```diff
@@ -16,28 +16,31 @@ export function todosReducer(state: TodosState = defaultState, action: Action):
             return Object.assign({}, state, {todos: [... state.todos, action.payload]});
         }
 
-        case TodosActions.REM_TODO: {
+        case TodosActions.REMOVE_TODO: {
             return Object.assign({}, state, {todos: state.todos.filter((todo: Todo) => todo !== action.payload as Todo)});
         }
 
         case TodosActions.SET_DONE: {
             const {done, todo} = action.payload;
 
-            const targetTodoClone: Todo = Object.assign({}, state.todos.find((currentTodo: Todo) => currentTodo.id === todo.id), {done});
+            const doneTodo: Todo = Object.assign({}, state.todos.find((currentTodo: Todo) => currentTodo.id === todo.id), {done});
 
-            return Object.assign({}, state, {todos: state.todos.map(currentTodo => currentTodo.id === todo.id ? targetTodoClone : currentTodo)});
+            return Object.assign({}, state, {todos: state.todos.map(currentTodo => currentTodo.id === todo.id ? doneTodo : currentTodo)});
         }
 
         case TodosActions.SET_NAME: {
             const {todo, name} = action.payload;
 
-            const targetTodoClone: Todo = Object.assign({},  state.todos.find((currentTodo: Todo) => currentTodo.id === todo.id), {name});
+            const renamedTodo: Todo = Object.assign({},  state.todos.find((currentTodo: Todo) => currentTodo.id === todo.id), {name});
 
-            return Object.assign({}, state, {todos: state.todos.map(currentTodo => currentTodo.id === todo.id ? targetTodoClone : currentTodo)});
+            return Object.assign({}, state, {todos: state.todos.map(currentTodo => currentTodo.id === todo.id ? renamedTodo : currentTodo)});
+        }
+
+        case TodosActions.FETCH_TODOS_SUCCESS: {
+            return Object.assign({}, state, {todos: [... state.todos, action.payload]});
```

Comment:
action.payload is just a one element or array of elements? From what I see it should be an array, but then when you execute this [... state.todos, action.payload] you will get an array of arrays that doesn't seem to be right